[ao] Add method in ModelReport to generate visualizer

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81673
* #81647
* __->__ #81589
* #81523

Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Add method in ModelReport to generate visualization

Summary: We created a ModelReportVisualization class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualization = model_reporter.generate_visualization()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualization instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualization

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualization"


Summary: We created a ModelReportVisualization class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualization = model_reporter.generate_visualization()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualization instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualization

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/quantization/fx/test_model_report_fx.py:
```diff
@@ -1033,7 +1034,7 @@ def test_generate_report(self):
             # prepare and callibrate two different instances of same model
             # prepare the model
-            example_input = torch.randn(1, 3, 3, 3)
+            example_input = model_full.get_example_inputs()[0]
```
this should be a tuple of inputs (e.g. example_inputs = (torch.randn(1, 3, 3, 3),))

vspenubarthi replied:
is a tuple of inputs where we get it from the actual class

andrewor14 commented on torch/ao/quantization/fx/_model_report/model_report.py:
```diff
@@ -275,5 +298,78 @@ def generate_model_report(
+        """
+        # check if user has generated reports at least once
+        if len(self._generated_reports) == 0:
+            raise Exception("You need to generate reports before you generate visualizers")
```
nit: This sounds a bit too colloquial? Maybe something like "Unable to generate visualizers without first generating reports"

andrewor14 commented on torch/ao/quantization/fx/_model_report/model_report.py:
```diff
@@ -275,5 +298,78 @@ def generate_model_report(
+                    present_info: Dict = module_fqns_to_features[module_fqn]
+
+                    # merge them together into the new unioned dict
+                    # same features keys -> same info, so okay if override
```
Maybe we should add a safety check to make sure the infos are in fact the same for the same keys

vspenubarthi replied:
Added

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Add method in ModelReport to generate visualizer"


Summary: We created a ModelReportVisualizer class, and the primary
way it is envisioned that it is accessed is:

```
model_report_visualizer = model_reporter.generate_visualizer()
```

This method only works after reports have been generated and it takes in
the generated reports and reformats them to be ordered by module, into
the format required by the ModelReportVisualization. It then generates
the visualizer instance and returns that to the user.

Test Plan: python test/test_quantization.py TestFxModelReportClass.test_generate_visualizer

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g
